solutions/src/test/java/ru/sberbank/school/task10/ConcurrentFixedThreadPoolTest.java

```diff
@@ -0,0 +1,147 @@
+package ru.sberbank.school.task10;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ConcurrentFixedThreadPoolTest {
+    private int core = 5;
+    private ThreadPool threadPool;
+
+    @Before
+    public void init() {
+        threadPool = new ConcurrentFixedThreadPool(core);
+    }
+
+    private Runnable newRunnable50ms(AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+            try {
+                TimeUnit.MILLISECONDS.sleep(50);
+            } catch (InterruptedException ignore) {
+//                        e.printStackTrace();
+            }
+            synchronized (counter) {
+                System.out.println("Task: " + counter.incrementAndGet()
+                        + " - Thread: " + Thread.currentThread().getName());
+            }
+            latch.countDown();
+        };
+    }
+
+    private Callable<String> newCallable50ms(AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+            TimeUnit.MILLISECONDS.sleep(50);
+            String res;
+            synchronized (counter) {
+                res = "Task: " + counter.incrementAndGet() + " - Thread: " + Thread.currentThread().getName();
+            }
+            latch.countDown();
+            return res + " " + LocalDateTime.now().toString();
+        };
+    }
+
+
+    @Test(timeout = 500)
+    public void normalWork() throws InterruptedException {
+        final AtomicInteger counter = new AtomicInteger(0);
+        Thread.sleep(10);
+        int initialThreadSize = Thread.getAllStackTraces().keySet().size();
+        int nowThreadSize;
+
+        threadPool.start();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        CountDownLatch latch = new CountDownLatch(25);
+        for (int i = 0; i < 25; i++) {
+            threadPool.execute(newRunnable50ms(counter, latch));
+        }
+        latch.await();
+        assertEquals(25, counter.get());
+
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        threadPool.stopNow();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(nowThreadSize, initialThreadSize);
+    }
+
+    @Test(timeout = 500)
+    public void callableService() throws InterruptedException, ExecutionException {
+        final List<Future<String>> futureList = new ArrayList<>();
+        final AtomicInteger counter = new AtomicInteger(0);
+        int initialThreadSize = Thread.getAllStackTraces().keySet().size();
```

Comment:
Distance between variable &#39;initialThreadSize&#39; declaration and its first usage is 4, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-113040729>*